src/maliput_osm/builder/road_geometry_builder.cc

```diff
@@ -50,26 +50,34 @@ std::unique_ptr<const maliput::api::RoadGeometry> RoadGeometryBuilder::operator(
       .ScaleLength(builder_configuration_.scale_length)
       .InertialToBackendFrameTranslation(builder_configuration_.inertial_to_backend_frame_translation);
 
+  maliput_sparse::builder::BranchPointBuilder bp_builder = rg_builder.StartBranchPoints();
   for (const auto& osm_segment : osm_segments) {
     maliput_sparse::builder::JunctionBuilder junction_builder = rg_builder.StartJunction();
     junction_builder.Id(maliput::api::JunctionId{osm_segment.first});
     maliput_sparse::builder::SegmentBuilder segment_builder = junction_builder.StartSegment();
     segment_builder.Id(maliput::api::SegmentId{osm_segment.first});
```

Comment:
I have to modify this to create more of one segment per junction when corresponds.

The order of creation is junction-> segment->lane 
However, in order to know how many segments should be added to the junction I should look into the lane's connections information (if more than 1 connection then the junction contains the ambiguity road connection)

We need to provide a method that analyzes the segments and suggests an organization for the junctions. Wdyt?

Reply:
I concur. I don't like too much the lazy initialization approach here.